packages/headless/src/controllers/core/facets/facet-dependencies-manager/headless-facet-dependencies-manager.ts

```diff
@@ -0,0 +1,194 @@
+  CategoryFacetSection,
+  DateFacetSection,
+  FacetOptionsSection,
+  FacetSection,
+  NumericFacetSection,
+} from '../../../../state/state-sections';
+import {loadReducerError} from '../../../../utils/errors';
+
+export interface AnyFacetDependency<T extends AnyFacetValueRequest> {
+  /**
+   * The id of the facet to watch.
+   */
+  parentFacetId: string;
+  /**
+   * The callback that determines whether this dependency is met
+   * @example
+   * ```ts
+   * const fileTypeHasSomeSelectedValue: AnyFacetDependency<FacetValueRequest> = {
+   *   parentFacetId: 'filetype',
+   *   isDependencyMet: (parentValues) => parentValues.some(value => value.state === 'selected')
+   * }
+   * ```
+   * .
+   * @param parentValues - The current values of the facet with the id equal to `parentFacetId`.
+   * @returns Whether the intended dependency is met.
+   */
+  isDependencyMet(parentValues: T[]): boolean;
+}
+
+export interface FacetDependenciesManagerProps {
+  /**
+   * The id of the facet to enable or disable depending on whether dependencies are met.
+   */
+  dependentFacetId: string;
+  /**
+   * The dependencies to watch.
+   *
+   * * If any of these dependencies are met, the dependent facet is enabled.
+   * * If none of these dependencies are met, the dependent facet is disabled.
+   */
+  dependencies: AnyFacetDependency<AnyFacetValueRequest>[];
+}
+
+export interface FacetDependenciesManager {
+  /**
+   * Stops enabling and disabling the dependent facet.
+   */
+  stopWatching(): void;
+}
+
+/**
+ * A manager that enables or disables a facet based on whether dependencies are met.
+ * @param engine - The headless engine.
+ * @param props - The configurable `FacetDependenciesManager` properties.
+ * @returns A new facet dependencies manager.
+ */
+export function buildFacetDependenciesManager(
+  engine: CoreEngine,
+  props: FacetDependenciesManagerProps
+): FacetDependenciesManager {
+  if (!loadFacetDependenciesManagerReducers(engine)) {
+    throw loadReducerError;
+  }
+  const lastObservedValuesPerParentFacet: Record<string, string> = {};
+
+  const isFacetEnabled = (facetId: string) => {
+    return engine.state.facetOptions.facets[facetId]?.enabled ?? true;
+  };
+
+  const getFacetValuesById = (facetId: string) => {
+    if (engine.state.facetSet && facetId in engine.state.facetSet) {
+      return engine.state.facetSet[facetId].currentValues;
+    }
+    if (
+      engine.state.categoryFacetSet &&
+      facetId in engine.state.categoryFacetSet
+    ) {
+      return engine.state.categoryFacetSet[facetId]!.request.currentValues;
+    }
+    if (
+      engine.state.numericFacetSet &&
+      facetId in engine.state.numericFacetSet
+    ) {
+      return engine.state.numericFacetSet[facetId].currentValues;
+    }
+    if (engine.state.dateFacetSet && facetId in engine.state.dateFacetSet) {
+      return engine.state.dateFacetSet[facetId].currentValues;
+    }
+    return null;
+  };
+
+  const updateLastObservedValues = () => {
```

Comment:
It's called update but it's mainly used to see if values were changed, I think the method name should reflect that